components/core/src/package/ident.rs

```diff
@@ -33,8 +37,31 @@ pub trait Identifiable: fmt::Display {
     fn fully_qualified(&self) -> bool { self.version().is_some() && self.release().is_some() }
 
     fn valid(&self) -> bool {
-        let re = Regex::new(r"^[A-Za-z0-9_-]+$").unwrap();
-        re.is_match(self.name())
+        if self.fully_qualified() {
+            IDENT_NAME_RE.is_match(self.name())
+            && !INOPERABLE_SEQUENCE_RE.is_match(self.version().unwrap())
+            && !INOPERABLE_SEQUENCE_RE.is_match(self.release().unwrap())
```

Comment:
Doesn't release have to be a timestamp string? Or is that a thing that we've always implicitly assumed, but never enforced, and now we have packages that have weird releases?